pkg/controller/seed-controller-manager/mla/resources.go

```diff
@@ -0,0 +1,353 @@
+	}
+  }
+
+  # read path - cluster-local access only
+  server {
+	listen             8081;
+	proxy_set_header   X-Scope-OrgID {{ .TenantID}};
+
+	# k8s probes
+	location = / {
+	  return 200 'OK';
+	  auth_basic off;
+	}
+
+	# location = /api/prom/tail {
+	#   proxy_pass       http://loki-distributed-querier.{{ .Namespace}}.svc.cluster.local:3100$request_uri;
+	#   proxy_set_header Upgrade $http_upgrade;
+	#   proxy_set_header Connection "upgrade";
+	# }
+
+	# location = /loki/api/v1/tail {
+	#   proxy_pass       http://loki-distributed-querier.{{ .Namespace}}.svc.cluster.local:3100$request_uri;
+	#   proxy_set_header Upgrade $http_upgrade;
+	#   proxy_set_header Connection "upgrade";
+	# }
+
+	location ~ /loki/api/.* {
+	  proxy_pass       http://loki-distributed-query-frontend.{{ .Namespace}}.svc.cluster.local:3100$request_uri;
+	}
+
+	# Cortex Config
+	location ~ /api/prom/.* {
+	  proxy_pass       http://cortex-query-frontend.{{ .Namespace}}.svc.cluster.local:8080$request_uri;
+	}
+  }
+
+  # public read and write path - used for alertmanager only
+  server {
+	listen             8082;
+	proxy_set_header   X-Scope-OrgID {{ .TenantID}};
+
+	# Alertmanager Config
+	location ~ /api/prom/alertmanager.* {
+	  proxy_pass      http://cortex-alertmanager.{{ .Namespace}}.svc.cluster.local:8080$request_uri;
+	}
+	location ~ /api/v1/alerts {
+	  proxy_pass      http://cortex-alertmanager.{{ .Namespace}}.svc.cluster.local:8080$request_uri;
+	}
+	location ~ /multitenant_alertmanager/status {
+	  proxy_pass      http://cortex-alertmanager.{{ .Namespace}}.svc.cluster.local:8080$request_uri;
+	}
+  }
+}
+`
+
+const (
+	gatewayName         = "mla-gateway"
+	gatewayAlertName    = "mla-gateway-alert"
+	gatewayExternalName = "mla-gateway-ext"
+)
+
+type configTemplateData struct {
+	Namespace string
+	TenantID  string
+}
+
+func renderTemplate(tpl string, data interface{}) (string, error) {
+	t, err := template.New("base").Funcs(sprig.TxtFuncMap()).Parse(tpl)
+	if err != nil {
+		return "", fmt.Errorf("failed to parse as Go template: %v", err)
+	}
+
+	output := bytes.Buffer{}
+	if err := t.Execute(&output, data); err != nil {
+		return "", fmt.Errorf("failed to render template: %v", err)
+	}
+
+	return strings.TrimSpace(output.String()), nil
+}
+
+func GatewayConfigMapCreator(c *kubermaticv1.Cluster) reconciling.NamedConfigMapCreatorGetter {
+	return func() (string, reconciling.ConfigMapCreator) {
+		return gatewayName, func(cm *corev1.ConfigMap) (*corev1.ConfigMap, error) {
+			if cm.Data == nil {
+				configData := configTemplateData{
+					Namespace: resources.MLANamespace,
+					TenantID:  c.Name,
+				}
+				config, err := renderTemplate(nginxConfig, configData)
+				if err != nil {
+					return nil, fmt.Errorf("failed to render Prometheus config: %v", err)
+				}
+
+				cm.Data = map[string]string{
+					"nginx.conf": config}
+			}
+			return cm, nil
+		}
+	}
+}
+
+func GatewayAlertServiceCreator() reconciling.NamedServiceCreatorGetter {
```

Comment:
The service is missing selectors - the traffic will not be routed towards the Gateway

Reply:
done